### ISSUE TITLE
Should SyncBN be used as long as multi-gpu mode

### ISSUE THREAD

**Issue by levan92** [open-mmlab/mmdetection#7341]
Hi, can I check if it is recommended that `SyncBN` should be used over `BN` as long as we are using multi-gpu and using DDP? 

In multi-gpu mode, does using `SyncBN` vs `BN` affect how the LR should be scaled (in the linear scale rule)? 

The LRs in the given configs, let's say `configs/common/mstrain-poly_3x_coco_instance.py`, assumes `samples_per_gpu=2` for 4 GPUs, meaning an effective Batch Size of 8, with `BN` norm right?

**Comment by Czm369** [comment]
In my view, `SyncBN` may not be necessary for multi-gpu mode, but may be  necessary for multi-node mode. LR should be scaled with batch size (in the linear scale rule). If samples_per_gpu=2 for 4 GPUs, which means batchsize=8 , the LR = 0.02/2.

**Comment by levan92** [comment]
Looking through this discussion: https://discuss.pytorch.org/t/training-performance-degrades-with-distributeddataparallel/47152, it seems to be implied that SyncBN should be used in DDP mode (when each gpu has its own process), in order for the batch statistics to be calculated across the entire batch through all the gpus/processes? Although, also in the discussion, people have empirically found that DDP SyncBN gave slightly worse performance than just DP (without SyncBN). 

Does the open-mmlab team or anyone here have any insights on this?

**Comment by Czm369** [comment]
Since the detection model is usually large and the input image resolution is high, this will result in a small batch of the detection model, which will make the variance of the statistics calculated by BatchNorm during the training process very large and not as stable as the statistics obtained during the pre-training of the backbone network . Therefore, the norm_eval=True mode is generally used in training, and the BatchNorm statistics in the pre-trained backbone network are directly used. The few algorithms that use large batches are the norm_eval=False mode, such as NASFPN. For the backbone network without ImageNet pre-training and the batch is relatively small, you can consider using SyncBN.
